figures/filters.py

```diff
@@ -107,13 +110,25 @@ def filter_course_id(self, queryset, name, value):  # pylint: disable=unused-arg
         Query parameters with plus signs '+' in the string are automatically
         replaced with spaces, so we need to put the '+' back in for CourseKey
         to be able to create a course key object from the string
-        '''
+        """
         course_key = CourseKey.from_string(value.replace(' ', '+'))
         return queryset.filter(course_id=course_key)
 
+    def filter_user_username(self, queryset, name, value):  # pylint: disable=unused-argument
+        """
+        Filter by User's username
+        """
+        return queryset.filter(user__username=value)
+
+    def filter_user_fullname(self, queryset, name, value):  # pylint: disable=unused-argument
+        """
+        Filter by User's full name
+        """
+        return queryset.filter(user__profile__name=value)
```

Comment:
@asadmanzoor93 do we have index on `name` file of profile model? If not we should not create a fullname filter

Reply:
@ziafazal Yes, it is indexed.